File: String/codeforces_1917B.cpp

// Problem link ===>> https://codeforces.com/problemset/problem/1917/B
// submission link ===>> https://codeforces.com/contest/1917/submission/241487510

#include<bits/stdc++.h>
using namespace std;

int main(){

    int t;
    cin >> t;
    while(t--){
        int n;
        cin >> n;
        string word;
        cin >> word;

        set<char> st;
        int count = 0;

        for(auto it:word){
            st.insert(it);
            count += st.size();
        }

        cout << count << endl;
    }

    return 0;
}

File: String/codeforces_1917B.py

# Problem link ===>> https://codeforces.com/problemset/problem/1917/B

import sys


def count_distinct_substrings(word):
    seen_chars = set()
    count = 0
    for char in word:
        seen_chars.add(char)
        count += len(seen_chars)
    return count


if __name__ == "__main__":
    t = int(sys.stdin.readline())
    for _ in range(t):
        length = int(sys.stdin.readline())
        word = sys.stdin.readline().strip()
        result = count_distinct_substrings(word)
        print(result)
